pkg/koordlet/metriccache/metric_cache.go

```diff
@@ -87,14 +88,18 @@ type MetricCache interface {
 	InsertContainerResourceMetric(t time.Time, containerResUsed *ContainerResourceMetric) error
 	InsertNodeCPUInfo(info *NodeCPUInfo) error
 	InsertNodeLocalStorageInfo(info *NodeLocalStorageInfo) error
+	InsertNodeDeviceInfo(device Devices) error
 	InsertBECPUResourceMetric(t time.Time, metric *BECPUResourceMetric) error
 	InsertContainerInterferenceMetrics(t time.Time, metric *ContainerInterferenceMetric) error
 	InsertPodInterferenceMetrics(t time.Time, metric *PodInterferenceMetric) error
+
+	GetNodeMetricAggregateResult(param *QueryParam) (NodeMetricsAggregateResult, error)
```

Comment:
it is not a good idea to add more and more interfaces into MetricCache, metric cache should just provide basic query method. So it's better to move GetNodeMetricAggregateResult out as a util function in statesinformer pkg.